src/main/java/edu/harvard/iq/dataverse/UserServiceBean.java

```diff
@@ -34,5 +41,368 @@ public AuthenticatedUser save( AuthenticatedUser user ) {
+     * - With Role added as a transient field
+     * @param searchTerm
+     * @param sortKey
+     * @param resultLimit
+     * @param offset
+     * @return 
+     */
+    public List<AuthenticatedUser> getAuthenticatedUserList(String searchTerm, String sortKey, Integer resultLimit, Integer offset){
+        
+        if ((offset == null)||(offset < 0)){
+            offset = 0;
+        }
+        
+        List<Object[]> userResults = getUserListCore(searchTerm, sortKey, resultLimit, offset);
+        
+        // Initialize empty list for AuthenticatedUser objects
+        //
+        List<AuthenticatedUser> viewObjects = new ArrayList<>();
+        
+        if (userResults == null){
+            return viewObjects;
+        }
+
+        // -------------------------------------------------
+        // GATHER GIANT HASHMAP OF ALL { user identifier : [role, role, role] }
+        // -------------------------------------------------
+
+        // Iterate through results, retrieving only the assignee identifiers
+        // Note: userInfo[1], the assigneeIdentifier, cannot be null in the database
+        //
+        List<String> identifiers = userResults.stream()
+                                        .map(userInfo -> (String)userInfo[1])
+                                        .collect(Collectors.toList())
+                                       ;
+  
+        HashMap<String, List<String>> roleLookup = retrieveRolesForUsers(identifiers);
+        if (roleLookup == null){
+            roleLookup = new HashMap<>();
+        }
+        //  1st Loop : 
+        // gather  [ @user, .....] 
+        // get the hashmap
+        
+        
+        // -------------------------------------------------
+        // We have results, format them into AuthenticatedUser objects
+        // -------------------------------------------------
+        int rowNum = offset++;   // used for the rowNumber
+        String roleString;
+        for (Object[] userInfo : userResults) {  
+            // GET ROLES FOR THIS USER FROM GIANT HASHMAP
+            rowNum++;
+            
+            //String roles = getUserRolesAsString((Integer) dbResultRow[0]);
+            roleString = "";
+            List<String> roleList = roleLookup.get("@" + (String)userInfo[1]);
+            if ((roleList != null)&&(!roleList.isEmpty())){
+                roleString = roleList.stream().collect(Collectors.joining(", "));
+            }
+            AuthenticatedUser singleUser = createAuthenticatedUserForView(userInfo, roleString, rowNum);            
+            viewObjects.add(singleUser);
+        }
+        
+        
+        return viewObjects;
+    }
+    
+    private AuthenticatedUser createAuthenticatedUserForView (Object[] dbRowValues, String roles, int rowNum){
+        AuthenticatedUser user = new AuthenticatedUser();
+        user.setRowNum(rowNum);
+        user.setId(new Long((Integer)dbRowValues[0]));
```

Comment:
e.g. can cast to an `int` but not `long`, not sure why.  changed to `user.setId(new Long((int)dbRowValues[0]));`